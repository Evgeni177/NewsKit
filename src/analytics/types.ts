export type TealiumLinkEvent = {
  event_navigation_action: React.ReactNode;
  event_navigation_browsing_method?: string;
};

export type TealiumViewEvent = {
  page_url: string;
};

export type InteractionEvent = {
  type: 'link';
  data: TealiumLinkEvent;
};

export type PageViewEvent = {
  type: 'view';
  data: TealiumViewEvent;
};

export type AnalyticsEvent = InteractionEvent | PageViewEvent;

export type UTag = {
  link(e: TealiumLinkEvent): void;
  view(e: TealiumViewEvent): void;
};

export interface ExtendedWindow extends Window {
  utag: UTag;
}
