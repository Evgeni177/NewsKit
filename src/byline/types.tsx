import {SizingKeys, SpacePresetKeys} from '../theme';

export interface BylineData {
  author: string;
  href?: string;
  title?: string;
  location?: string;
  ariaLabel?: string;
}

export interface BylineProps {
  bylineData: BylineData[];
  overrides?: {
    stylePreset?: string;
    typePreset?: string;
    space?: SizingKeys;
    link?: {
      stylePreset?: string;
      typePreset?: string;
    };
    divider?: {
      stylePreset?: string;
      spaceInline?: SpacePresetKeys;
    };
  };
}
