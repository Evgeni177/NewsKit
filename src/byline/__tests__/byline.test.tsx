import {Byline} from '../byline';
import {BylineProps} from '../types';
import {renderToFragmentWithTheme} from '../../test/test-utils';
import {createTheme, compileTheme} from '../../theme';

describe('Byline', () => {
  test('renders correctly with one author', () => {
    const props: BylineProps = {
      bylineData: [
        {
          author: 'Jane Smith',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Byline, props) as any;

    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with multiple authors', () => {
    const props: BylineProps = {
      bylineData: [
        {
          author: 'Jane Smith',
        },
        {
          author: 'John Smith',
        },
        {
          author: 'Sam Smith',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Byline, props) as any;

    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with authors and links', () => {
    const props: BylineProps = {
      bylineData: [
        {
          author: 'Jane Smith',
          href: 'www.example.com',
          location: 'London',
        },
        {
          author: 'John Smith',
          href: 'www.example.com',
        },
        {
          author: 'John Smith',
          href: 'www.example.com',
          title: 'Developer manager',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Byline, props) as any;

    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly multiple authors with link only', () => {
    const props: BylineProps = {
      bylineData: [
        {
          author: 'Jane Smith',
          href: 'www.example.com',
        },
        {
          author: 'John Smith',
          href: 'www.example.com',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Byline, props) as any;

    expect(fragment).toMatchSnapshot();
  });

  test("renders correctly with author's name, author's information and no href provided", () => {
    const props: BylineProps = {
      bylineData: [
        {
          author: 'Jane Smith',
          title: 'Author',
        },
        {
          author: 'John Smith',
          location: 'Milan',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Byline, props) as any;

    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with authors, location, titles and links', () => {
    const props: BylineProps = {
      bylineData: [
        {
          author: 'Jane Smith',
          title: 'Author',
          location: 'London',
          href: 'www.example.com',
        },
        {
          author: 'John Smith',
          title: 'Other Author',
          location: 'London',
          href: 'www.example.com',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Byline, props) as any;

    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with overrides', () => {
    const myCustomTheme = createTheme({
      name: 'my-custom-byline-theme',
      overrides: {
        stylePresets: {
          bylineCustom: {
            base: {
              color: '#f000dc',
            },
          },
          bylineLinkCustom: {
            base: {
              color: '#d82059',
            },
          },
          bylineDividerCustom: {
            base: {
              color: '#27a727',
            },
          },
        },
      },
    });

    const props: BylineProps = {
      overrides: {
        stylePreset: 'bylineCustom',
        typographyPreset: 'label040',
        space: 'sizing030',
        link: {
          stylePreset: 'bylineLinkCustom',
          typographyPreset: 'label040',
        },
        divider: {
          stylePreset: 'bylineDividerCustom',
          spaceInline: 'space030',
        },
      },
      bylineData: [
        {
          author: 'Jane Smith',
          title: 'Author',
          location: 'London',
          href: 'www.example.com',
        },
        {
          author: 'John Smith',
          title: 'Other Author',
          location: 'London',
          href: 'www.example.com',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(
      Byline,
      props,
      compileTheme(myCustomTheme),
    ) as any;

    expect(fragment).toMatchSnapshot();
  });
});
