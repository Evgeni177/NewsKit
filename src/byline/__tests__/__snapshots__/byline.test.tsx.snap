// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Byline renders correctly multiple authors with link only 1`] = `
<DocumentFragment>
  .emotion-12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: calc(-8px/2);
  margin-bottom: calc(-8px/2);
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: calc(8px/2);
  margin-bottom: calc(8px/2);
}

.emotion-4 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
}

.emotion-4 svg {
  fill: #535353;
}

.emotion-4::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-4::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-5 {
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-3 {
  margin-left: 8px;
  color: #c0c0c0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-3 svg {
  fill: #c0c0c0;
}

.emotion-2 {
  display: inline-block;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  color: #0a68c1;
}

.emotion-2 svg {
  fill: #0a68c1;
}

.emotion-2:hover:not(:disabled) {
  color: #02509b;
}

.emotion-2:hover:not(:disabled) svg {
  fill: #02509b;
}

.emotion-2:active:not(:disabled) {
  color: #01396e;
}

.emotion-2:active:not(:disabled) svg {
  fill: #01396e;
}

.emotion-2:visited:not(:disabled) {
  color: #7547a3;
}

.emotion-2:visited:not(:disabled) svg {
  fill: #7547a3;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    vertical-align: text-bottom;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
  display: block;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-12"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-5"
      >
        <span
          class="emotion-4"
        >
          <a
            aria-label="more by Jane Smith"
            class="emotion-2"
            data-author="Jane Smith"
            href="www.example.com"
            rel="author"
          >
            <span
              class="emotion-1"
            >
              <span
                class="emotion-0"
              >
                Jane Smith
              </span>
            </span>
          </a>
          <span
            class="emotion-3"
          >
            |
          </span>
        </span>
      </div>
    </div>
    <div
      class="emotion-6"
    >
      <span
        class="emotion-4"
      >
        <a
          aria-label="more by John Smith"
          class="emotion-2"
          data-author="John Smith"
          href="www.example.com"
          rel="author"
        >
          <span
            class="emotion-1"
          >
            <span
              class="emotion-0"
            >
              John Smith
            </span>
          </span>
        </a>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Byline renders correctly with author's name, author's information and no href provided 1`] = `
<DocumentFragment>
  .emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: calc(-8px/2);
  margin-bottom: calc(-8px/2);
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: calc(8px/2);
  margin-bottom: calc(8px/2);
}

.emotion-1 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
}

.emotion-1 svg {
  fill: #535353;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-2 {
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  margin-left: 8px;
  color: #c0c0c0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 svg {
  fill: #c0c0c0;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-3"
    >
      <div
        class="emotion-2"
      >
        <span
          class="emotion-1"
        >
          Jane Smith, Author
          <span
            class="emotion-0"
          >
            |
          </span>
        </span>
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <span
        class="emotion-1"
      >
        John Smith, Milan
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Byline renders correctly with authors and links 1`] = `
<DocumentFragment>
  .emotion-21 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: calc(-8px/2);
  margin-bottom: calc(-8px/2);
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: calc(8px/2);
  margin-bottom: calc(8px/2);
}

.emotion-3 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
}

.emotion-3 svg {
  fill: #535353;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-6 {
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-4 {
  margin-left: 8px;
  color: #c0c0c0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-4 svg {
  fill: #c0c0c0;
}

.emotion-2 {
  display: inline-block;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  color: #0a68c1;
}

.emotion-2 svg {
  fill: #0a68c1;
}

.emotion-2:hover:not(:disabled) {
  color: #02509b;
}

.emotion-2:hover:not(:disabled) svg {
  fill: #02509b;
}

.emotion-2:active:not(:disabled) {
  color: #01396e;
}

.emotion-2:active:not(:disabled) svg {
  fill: #01396e;
}

.emotion-2:visited:not(:disabled) {
  color: #7547a3;
}

.emotion-2:visited:not(:disabled) svg {
  fill: #7547a3;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    vertical-align: text-bottom;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
  display: block;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-21"
  >
    <div
      class="emotion-7"
    >
      <div
        class="emotion-6"
      >
        <span
          class="emotion-3"
        >
          <a
            aria-label="more by Jane Smith"
            class="emotion-2"
            data-author="Jane Smith"
            href="www.example.com"
            rel="author"
          >
            <span
              class="emotion-1"
            >
              <span
                class="emotion-0"
              >
                Jane Smith
              </span>
            </span>
          </a>
          <span
            aria-hidden="true"
            class="emotion-3"
          >
            , 
          </span>
          London
          <span
            class="emotion-4"
          >
            |
          </span>
        </span>
      </div>
    </div>
    <div
      class="emotion-7"
    >
      <div
        class="emotion-6"
      >
        <span
          class="emotion-3"
        >
          <a
            aria-label="more by John Smith"
            class="emotion-2"
            data-author="John Smith"
            href="www.example.com"
            rel="author"
          >
            <span
              class="emotion-1"
            >
              <span
                class="emotion-0"
              >
                John Smith
              </span>
            </span>
          </a>
          <span
            class="emotion-4"
          >
            |
          </span>
        </span>
      </div>
    </div>
    <div
      class="emotion-7"
    >
      <span
        class="emotion-3"
      >
        <a
          aria-label="more by John Smith"
          class="emotion-2"
          data-author="John Smith"
          href="www.example.com"
          rel="author"
        >
          <span
            class="emotion-1"
          >
            <span
              class="emotion-0"
            >
              John Smith
            </span>
          </span>
        </a>
        <span
          aria-hidden="true"
          class="emotion-3"
        >
          , 
        </span>
        Developer manager
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Byline renders correctly with authors, location, titles and links 1`] = `
<DocumentFragment>
  .emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: calc(-8px/2);
  margin-bottom: calc(-8px/2);
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: calc(8px/2);
  margin-bottom: calc(8px/2);
}

.emotion-3 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
}

.emotion-3 svg {
  fill: #535353;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-6 {
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-4 {
  margin-left: 8px;
  color: #c0c0c0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-4 svg {
  fill: #c0c0c0;
}

.emotion-2 {
  display: inline-block;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  color: #0a68c1;
}

.emotion-2 svg {
  fill: #0a68c1;
}

.emotion-2:hover:not(:disabled) {
  color: #02509b;
}

.emotion-2:hover:not(:disabled) svg {
  fill: #02509b;
}

.emotion-2:active:not(:disabled) {
  color: #01396e;
}

.emotion-2:active:not(:disabled) svg {
  fill: #01396e;
}

.emotion-2:visited:not(:disabled) {
  color: #7547a3;
}

.emotion-2:visited:not(:disabled) svg {
  fill: #7547a3;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    vertical-align: text-bottom;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
  display: block;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-14"
  >
    <div
      class="emotion-7"
    >
      <div
        class="emotion-6"
      >
        <span
          class="emotion-3"
        >
          <a
            aria-label="more by Jane Smith"
            class="emotion-2"
            data-author="Jane Smith"
            href="www.example.com"
            rel="author"
          >
            <span
              class="emotion-1"
            >
              <span
                class="emotion-0"
              >
                Jane Smith
              </span>
            </span>
          </a>
          <span
            aria-hidden="true"
            class="emotion-3"
          >
            , 
          </span>
          Author, London
          <span
            class="emotion-4"
          >
            |
          </span>
        </span>
      </div>
    </div>
    <div
      class="emotion-7"
    >
      <span
        class="emotion-3"
      >
        <a
          aria-label="more by John Smith"
          class="emotion-2"
          data-author="John Smith"
          href="www.example.com"
          rel="author"
        >
          <span
            class="emotion-1"
          >
            <span
              class="emotion-0"
            >
              John Smith
            </span>
          </span>
        </a>
        <span
          aria-hidden="true"
          class="emotion-3"
        >
          , 
        </span>
        Other Author, London
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Byline renders correctly with multiple authors 1`] = `
<DocumentFragment>
  .emotion-10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: calc(-8px/2);
  margin-bottom: calc(-8px/2);
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: calc(8px/2);
  margin-bottom: calc(8px/2);
}

.emotion-1 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
}

.emotion-1 svg {
  fill: #535353;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-2 {
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  margin-left: 8px;
  color: #c0c0c0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 svg {
  fill: #c0c0c0;
}

<div
    class="emotion-10"
  >
    <div
      class="emotion-3"
    >
      <div
        class="emotion-2"
      >
        <span
          class="emotion-1"
        >
          Jane Smith
          <span
            class="emotion-0"
          >
            |
          </span>
        </span>
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-2"
      >
        <span
          class="emotion-1"
        >
          John Smith
          <span
            class="emotion-0"
          >
            |
          </span>
        </span>
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <span
        class="emotion-1"
      >
        Sam Smith
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Byline renders correctly with one author 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: calc(-8px/2);
  margin-bottom: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: calc(8px/2);
  margin-bottom: calc(8px/2);
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
}

.emotion-0 svg {
  fill: #535353;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-2"
  >
    <div
      class="emotion-1"
    >
      <span
        class="emotion-0"
      >
        Jane Smith
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Byline renders correctly with overrides 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: calc(-12px/2);
  margin-bottom: calc(-12px/2);
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-top: calc(12px/2);
  margin-bottom: calc(12px/2);
}

.emotion-6 {
  margin-right: 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-3 {
  margin: 0;
  padding: 1px 0;
  color: #f000dc;
  font-family: "Noto Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.4em;
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4625em;
}

.emotion-2 {
  display: inline-block;
  -webkit-transition-property: color;
  transition-property: color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  color: #d82059;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    vertical-align: text-bottom;
  }
}

.emotion-0 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: inline-block;
  display: block;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.4em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4625em;
}

.emotion-4 {
  margin-left: 12px;
  color: #27a727;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
    class="emotion-14"
  >
    <div
      class="emotion-7"
    >
      <div
        class="emotion-6"
      >
        <span
          class="emotion-3"
        >
          <a
            aria-label="more by Jane Smith"
            class="emotion-2"
            data-author="Jane Smith"
            href="www.example.com"
            rel="author"
          >
            <span
              class="emotion-1"
            >
              <span
                class="emotion-0"
              >
                Jane Smith
              </span>
            </span>
          </a>
          <span
            aria-hidden="true"
            class="emotion-3"
          >
            , 
          </span>
          Author, London
          <span
            class="emotion-4"
          >
            |
          </span>
        </span>
      </div>
    </div>
    <div
      class="emotion-7"
    >
      <span
        class="emotion-3"
      >
        <a
          aria-label="more by John Smith"
          class="emotion-2"
          data-author="John Smith"
          href="www.example.com"
          rel="author"
        >
          <span
            class="emotion-1"
          >
            <span
              class="emotion-0"
            >
              John Smith
            </span>
          </span>
        </a>
        <span
          aria-hidden="true"
          class="emotion-3"
        >
          , 
        </span>
        Other Author, London
      </span>
    </div>
  </div>
</DocumentFragment>
`;
